docs/codejail.rst

```diff
@@ -0,0 +1,43 @@
+Codejail service
+################
+
+The ``codejail`` devstack component (codejail-service) requires some additional configuration before it can be enabled. This page describes how to set it up and debug it.
+
+Background
+**********
+
+Both LMS and CMS can run Python code submitted by instructors and learners in order to implement custom Python-graded problems. By default this involves running the code on the same host as edxapp itself. Ordinarily this would be quite dangerous, but we use a sandboxing library called `codejail <https://github.com/openedx/codejail>`__ in order to confine the code execution in terms of disk and network access as well as memory, CPU, and other resource limits. Part of these restrictions are implemented via AppArmor, a utility available in some Linux distributions (including Debian and Ubuntu).
+
+While AppArmor provides good protection, a sandbox escape could still be possible due to misconfiguration or bugs in AppArmor. For defense in depth, we're setting up a dedicated `codejail service <https://github.com/openedx/codejail-service>`__ that will perform code execution for edxapp and which will allow further isolation.
+
+The default edxapp codejail defaults to unsafe, direct execution of Python code, and this remains true in devstack. We don't even have a way to run on-host codejail securely in devstack. In constrast, the codejail service refuses to run if codejail has not been configured properly, and we've included a way to run it in devstack.
+
+Configuration
+*************
+
+In order to run the codejail devstack component:
+
+1. Install AppArmor: ``sudo apt install apparmor``
```

Comment:
- Is this directly on my local machine (Mac, etc.)? Could you make that more explicit?
- Once I'm clear, I can run through this as well.

Reply:
Yes, it's meant to be run on the host machine. Although now I'm realizing that I'm not actually sure where you're supposed to run it on a Mac! It's a Linux command, and won't work for you.

How does this all work on Mac, anyhow? Maybe there's a Linux VM that all the docker stuff runs inside? If so, this would need to be run in that Linux VM.

Reply:
Maybe this will help us? https://github.com/abiosoft/colima/discussions/1019

Reply:
I don't see anything promising in there. My understanding is that Colima sets up an Ubuntu-based Linux VM and runs docker inside of that, and that to get this working on a Mac you'd have to access that VM directly and install apparmor (and this profile) inside of the VM. I see references on the web to `colima ssh`, which might help?

This is definitely going to be harder to run on a Mac.